src/ssl.c

```diff
@@ -1993,6 +1993,15 @@ int wolfSSL_dtls_set_peer(WOLFSSL* ssl, void* peer, unsigned int peerSz)
         return WOLFSSL_SUCCESS;
     }
 
+#if !defined(WOLFSSL_IPV6) && defined(HAVE_SOCKADDR)
+    {
+        SOCKADDR *addr;
+        addr = (SOCKADDR*)peer;
+        if (peerSz < sizeof(addr->sa_family) || addr->sa_family == AF_INET6)
+            return WOLFSSL_FAILURE;
+    }
+#endif
```

Comment:
What if the `peer` is a custom address and not a `SOCKADDR`? Currently we don't require it to be a `SOCKADDR` and a user can user a custom address with a custom I/O callback.